src/common/datatypes.xsl

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:m="http://csrc.nist.gov/ns/oscal/metaschema/1.0"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                version="3.0">
+    <!--
+         Normalize deprecated datatype names
+         ex. m:datatype-normalize('dateTime-with-timezone') => date-time-with-timezone
+         Note: unknown and valid datatypes will be passed along as-is
+    -->
+    <xsl:function
+        name="m:datatype-normalize" as="xs:string">
+        <xsl:param name="raw-datatype" as="xs:string" />
+        <xsl:variable name="deprecation-map">
+            <!-- these old names are permitted for now, while only deprecated -->
+            <!--metaschema/schema/xml/metaschema.xsd
+                 line 1052 inside  /*/xs:simpleType[@name='SimpleDatatypesType']> -->
+            <entry key="base64Binary">base64</entry>
+            <entry key="dateTime">date-time</entry>
+            <entry key="dateTime-with-timezone">date-time-with-timezone</entry>
+            <entry key="email">email-address</entry>
+            <entry key="nonNegativeInteger">non-negative-integer</entry>
+            <entry key="positiveInteger">positive-integer</entry>
+        </xsl:variable>
+        <xsl:choose>
+            <xsl:when test="$deprecation-map/entry[@key=$raw-datatype]">
+                <xsl:value-of select="$deprecation-map/entry[@key=$raw-datatype]"/>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:value-of select="$raw-datatype"/>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:function>
+    
+    <!-- used in documentation generation pipelines -->
+    <xsl:variable name="metaschema-reference-url" as="xs:string">https://pages.nist.gov/metaschema/specification/datatypes</xsl:variable>
```

Comment:
Minor, non-blocking: would it be to our advantage to make it a [`param`](https://developer.mozilla.org/en-US/docs/Web/XSLT/Element/param) so it can be overwritten for local testing (override to `http://localhost:$PORT/metaschema` to Dmitry's discussion about this during the prior week) or if the production URL changes again without needing to change the "library" implementation in this datatypes stylesheet as imported.

@wendellpiez, thoughts?

```suggestion
    <xsl:param name="metaschema-reference-url" as="xs:string">https://pages.nist.gov/metaschema/specification/datatypes</xsl:variable>
```

Reply:
Sure, go for it. 👍